### PR TITLE
Enable Background effects!

### PR DESCRIPTION
**Pull request by Hamms**
video

(actual video quality is much higher, this is just the result of a quick video-to-gif conversion)

Note that the two background effects that rely on SVG generation are not yet at 100%; the svg doesn't load, so they just show colors.

### CODE REVIEW

File: src/generateData.js [hunk 1]
```diff
@@ -6,6 +6,16 @@ const ANIMATION_COUNT = danceParty.constants.MOVE_NAMES.length;
 const FRAMES_PER_ANIMATION = danceParty.constants.FRAMES;
 const SPRITE_NAMES = danceParty.constants.SPRITE_NAMES;
 
+// inferred from /dashboard/config/blocks/Dancelab/Dancelab_setBackgroundEffect.json
+const BG_EFFECT_NAMES = [
```
Comment (joshlory) [inline src/generateData.js]: Should these be exported from https://github.com/code-dot-org/dance-party/blob/master/src/constants.js?

File: src/replayToMovie.js [hunk 2]
```diff
@@ -55,13 +56,18 @@ require('@code-dot-org/p5.play/lib/p5.play');
 
 const p5Inst = new P5(function (p5obj) {
   p5obj._fixedSpriteAnimationFrameSizes = true;
+  p5obj.width = WIDTH;
+  p5obj.height = HEIGHT;
 });
```
Comment (joshlory) [inline src/replayToMovie.js]: Can we get updated to use `loadP5.js`?

File: src/replayToMovie.js [hunk 3]
```diff
@@ -144,8 +150,15 @@ module.exports.runTestExport = async (outputPath, replay) => {
 module.exports.renderImages = async (replay, writer) => {
   const sprites = [];
   for (const frame of replay) {
-    for (let i = 0; i < frame.length; i++) {
-      const entry = frame[i];
+    // temporarily support both the new version of replay logs that contain
+    // sprites as well as envrionmental data, and the old version that contains
+    // just sprites
```
Comment (joshlory) [inline src/replayToMovie.js]: Is the log format specification documented anywhere?  Will get really hard to change once we get start getting thousands of logs uploaded to S3.

(I guess we could make a breaking change, and no longer be able to process old logs that already have videos.)

File: src/replayToMovie.js (hunk 2 above)
Comment (Hamms) [reply src/replayToMovie.js]: unfortunately not without significant changes; I explored that, but we're using two different versions of Canvas in this and in the dance-party repo, with two different sets of requirements for the workaround.

I'd definitely like to explore unifying them at some point, but likely after HoC

File: src/replayToMovie.js (hunk 3 above)
Comment (Hamms) [reply src/replayToMovie.js]: Good point. It isn't, but it should be.

I'm planning on implementing this transition as a breaking change; from the "draft" specification to the true v1. Future changes will definitely have to be handled better.

File: src/generateData.js (hunk 1 above)
Comment (Hamms) [reply src/generateData.js]: Probably; given that this is just a QOL data gen script, I'm not considering that a high priority right now

**Pull request merged**